### ISSUE TITLE
Evaluation order wrong with transform macro

### ISSUE THREAD

**Issue by nklein** [wmannis/cl-svg#16]
The transform macro evaluates the body before evaluating the transforms. This means that code with affects the transforms is hard to reason about. The following example looks like it should put one circle at (0, 10) and one circle at (0,20) but it puts both circles at (0,20):
```
(let ((yoff 10))
  (cl-svg:transform ((cl-svg:translate 0 yoff))
    (prog1
        (cl-svg:draw scene (:circle :cx 0 :cx 0 :r 5))
      (incf yoff 10)))
  (cl-svg:transform ((cl-svg:translate 0 yoff))
    (cl-svg:draw scene (:circle :cx 0 :cx 0 :r 5))))
```

Here is a rewrite of the transform macro that honors the apparent evaluation order:
```
(defmacro transform ((&rest transformations) &body element)
  (if (atom (first transformations))
      (let ((trans (gensym)))
        `(let ((,trans ,transformations))
           (add-transform ,@element ,trans)))
      (let ((trans (gensym))
            (translist (gensym))
            (e (gensym)))
        `(let ((,translist (list ,@transformations))
               (,e ,@element))
           (dolist (,trans ,translist ,e)
             (add-transform ,e ,trans))))))
```

**Comment by nklein** [comment]
Added pull-request #19 for this issue.

**Comment by nklein** [comment]
Merged pull-request #19
